packages/core/src/classes/expression/base-expression-input.ts

```diff
@@ -160,13 +160,21 @@ export abstract class BaseExpressionInput {
       this.and().appendToExpression(`(${expression})`);
     }
 
-    Object.keys(names).forEach(nameKey => {
-      if (this.names[nameKey]) {
-        throw new Error(
-          `Failed to merge expression attribute names, there are multiple attributes names with key "${nameKey}"`
-        );
-      }
-    });
+    // THIS CHECK IS NOT NEEDED
```

Comment:
@andygonzalez2010 

From memory the reason this check was here was due to DynamoDB itself didn't support declaring multiple condition on same attribute within a expression..

With your below example... it would have created an incorrect query expression that would fail when executing against DynamoDB.

Not sure if this has changed since the time it was implemented. Would love to see some references proving this is now supported.

Reply:
im currently running this raw :
```
TransactWriteItems{
  "TransactItems": [
    {
      "Update": {
        "TableName": "test",
        "Key": {
          "PK": {
            "S": "UserId#111"
          },
          "SK": {
            "S": "RUNNING_TASKS"
          }
        },
        "ExpressionAttributeNames": {
          "#runningTaskIds": "runningTaskIds"
        },
        "ExpressionAttributeValues": {
          ":runningTaskIds": {
            "SS": ["12", "34"]
          },
          ":zero": {
            "N": "0"
          }
        },
        "ConditionExpression":
                "attribute_not_exists(#runningTaskIds) OR size(#runningTaskIds) = :zero",
        "UpdateExpression": "ADD #runningTaskIds :runningTaskIds"
      }
    }
  ]
}
```